src/pku2u/mod.rs

```diff
@@ -198,16 +214,21 @@ impl Pku2u {
 }
 
 impl Sspi for Pku2u {
+    #[cfg_attr(feature = "logging", instrument(level = "debug", ret, fields(state = self.state.as_ref()), skip_all))]
     fn complete_auth_token(&mut self, _token: &mut [SecurityBuffer]) -> Result<SecurityStatus> {
         Ok(SecurityStatus::Ok)
     }
 
+    #[cfg_attr(feature = "logging", instrument(level = "debug", ret, fields(state = self.state.as_ref()), skip(self, _flags)))]
     fn encrypt_message(
         &mut self,
         _flags: EncryptionFlags,
         message: &mut [SecurityBuffer],
         sequence_number: u32,
     ) -> Result<SecurityStatus> {
+        #[cfg(feature = "logging")]
+        trace!("{:?}", self.encryption_params);
```

Comment:
Tracing way would be:
```suggestion
        #[cfg(feature = "logging")]
        trace!(encryption_params = ?self.encryption_params);
```
(you don’t need a (format) message string at all)

Reply:
Fixed